### ISSUE TITLE
CA2235 string is not serializable

### ISSUE THREAD

**Issue by uandrei** [dotnet/roslyn-analyzers#1936]
#### Analyzer package

Microsoft.NetFramework.Analyzers

#### Package Version

Example: v2.6.2

#### Diagnostic ID

CA2235

#### Repro steps

```c#
[Serializable]
public class Log4NetSettings : ILog4NetSettings
{
	public Log4NetSettings()
	{
		this.GlobalContextProperties.Add("appRoot", Directory.GetCurrentDirectory());
		this.ConfigurationFilePath = "log4net.config";
		this.RootRepositoryName = "Root";
	}

	public IDictionary<string, object> GlobalContextProperties { get; } = new Dictionary<string, object>();

	public string ConfigurationFilePath { get; set; }

	public string RootRepositoryName { get; set; }

	public bool MonitorConfigurationFileChanges { get; set; }
}
```

#### Expected behavior

Build is successfull

#### Actual behavior

CA2235 complains that the string and bool properties are part of an object that is serializable, but string and bool are not serializable themselves. 

This issue doesn't happen in 2.6.0 version of the package

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn-analyzers/issues/1712
